_stbt/core.py

```diff
@@ -1178,18 +1187,38 @@ def wait_until(callable_, timeout_secs=10, interval_secs=0):
 
     We hope to solve both of the above drawbacks at some point in the future.
 
-    ``wait_until`` was added in stb-tester v22.
+    ``wait_until`` was added in stb-tester v22. The ``consecutive_secs``
+    parameter was added in v27.
     """
     import time
+
+    class NoValue(object):
+        pass
+
+    stable_value = NoValue
+    stable_since = None
     expiry_time = time.time() + timeout_secs
+
     while True:
         t = time.time()
         value = callable_()
-        if value:
+
+        if value != stable_value:
```

Comment:
If we're getting `Falsey` values I think this will compare them even if we're not interested in `consecutive_secs`.  I don't think this will be a problem for `FrameObject`s as all `False`y `FrameObject`s of the same concrete type are equal and only `is_visible` will be invoked, and the result of that should be cached.  I don't know if this could be a problem with other types.  Probably fine.